challenges/ai-ml/challenge-1.md

```diff
@@ -0,0 +1,250 @@
+
+Check [Systems Engineering Levels](../../levels/systems.pdf) for more details.
+
+# Interview Process
+
+The interview process will start with you receiving an invite to a private
+Slack channel. That channel will contain the interview panel. You can ask them
+about the engineering culture, work-life balance, or anything else that you
+would like to learn about Teleport.
+
+## Design Doc
+
+Before writing any actual code, we ask that you write a brief design document.
+The design document should cover: design approach, scope, proposed API, security
+considerations, CLI UX, and implementation details where appropriate. Start with
+a brief doc that covers the edge cases and design approach.
+
+Please submit the design document and all code in a GitHub repository. Public
+or private is your choice. Please submit the design document written in
+markdown as a Pull Request to allow us to provide you feedback on the proposed
+design.
+
+A few notes about the design document:
+
+* Try to get the design document approved within the first 2-3 days. This is to
+  ensure you have enough time to work on the implementation.
+* Avoid writing an overly detailed design document. Two to three pages is
+  sufficient.
+* Avoid sending us draft design documents. It is difficult to evaluate which
+  parts are draft and which parts are complete. Instead we encourage asking
+  questions in Slack and sharing a design document that is ready to be reviewed.
+
+Be sure to cover the following in your design:
+
+* Your approach to solving the challenge problem.
+* CLI user experience.
+
+Once the design document has been approved by two reviewers, move on to the
+implementation.
+
+## Implementation
+
+Split your code submission into roughly 2-3 Pull Requests to give the team an
+opportunity to review your code and provide feedback. Feel free to merge each
+PR after you have two approvals.
+
+Our team will do their best to provide a high quality review of the submitted
+Pull Requests in a reasonable time frame. You are spending your time on this,
+we are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using a
+"+1, -2" anonymous voting system: +1 is submitted whenever a team member
+accepts the submission, -2 otherwise.
+
+In case of a positive result, we will connect you to our HR and recruiting
+teams, who will work out the details and present an offer.
+
+In case of a negative score result, the hiring manager will contact you and share a
+list of the key observations from the team that affected the result.
+
+### Tools
+
+This task should be implemented in Python, Go or Rust and should work on 64-bit
+Linux or macOS machines.
+
+### Testing
+
+Key components of the challenge should have tests that cover the happy and
+unhappy scenarios. Do not try to achieve 100% test coverage as that will take
+too long.
+
+### Dependencies
+
+Please write as much of your own code as possible. Avoid relying on third
```

Comment:
We should also allow them to use pandas and other DS libraries. Maybe consider removing it

Reply:
Yeah, I'd expect to see the likes of Pandas/Scikit/Pytorch/Tensorflow being leveraged pretty heavily for this task.